### PR TITLE
Fix ADAL tokens, change UI for single/multiple revisions

### PR DESCRIPTION
**Pull request by nturinski**
Fixes https://github.com/microsoft/vscode-azurecontainerapps/issues/47

If the Container App is in single revision mode, the revision tree item shouldn't be displayed because it's just confusing.  Based on whether or not it's enabled, where the user can swap RevisionMode context action exists on different tree items (which seems like it could be potentially confusing).

The wonky stuff I'm doing with the credentials... Basically, the T2 package for registry will ignore the credentials and options that I pass it if `signRequest` is defined.  Since that command is only for backwards compatibility, the easiest way to fix this is to just make it undefined when we need to use it.   Since `parseContext` returns a new object, messing with credentials shouldn't delete `signRequest` everywhere.

### CODE REVIEW

File: package.nls.json
Comment (wwlorey) [inline package.nls.json:13]: What was the reasoning for this verbiage change?

**Review by wwlorey — COMMENTED** [review]
> The wonky stuff I'm doing with the credentials... Basically, the T2 package for registry will ignore the credentials and options that I pass it if signRequest is defined.

Did you file an issue on the SDK for this? This is a fine workaround for now but I could see other extensions having to do this same workaround.

I'm wondering if we made a mistake in keeping signRequest around for backwards compatibility 🤔  If this is just a bug on the SDK we're likely fine but maybe their behavior is intentional.

**Comment by nturinski** [comment]
> Did you file an issue on the SDK for this? This is a fine workaround for now but I could see other extensions having to do this same workaround.

Not yet, but I get the impression that it was pretty intentional.  They use a `coreAuth.isTokenCredential` command to check if signRequest is defined or not.  But this SDK also hard codes the default scope (with no option to add more) so I'm not really sure what they're trying to enforce here.

> I'm wondering if we made a mistake in keeping signRequest around for backwards compatibility 🤔 If this is just a bug on the SDK we're likely fine but maybe their behavior is intentional.

I don't think so.  I can imagine the backlash of a bunch of users who got our token and were upset that the 0.10.0 update _also_ broke all T1 SDK users which I imagine is practically everybody besides us.  As far as I can tell, the access token is the exact same so Azure really shouldn't care as long as it's only using the default scope.

**Comment by nturinski** [comment]
I do think that the ADAL token not working is a bug though.

```
if (coreAuth.isTokenCredential(credentialOrOptions)) {
            credential = credentialOrOptions;
            options = clientOptions;
        }
        else {
            options = credentialOrOptions !== null && credentialOrOptions !== void 0 ? credentialOrOptions : {};
        }
```

The SDK has a path for when the credential isn't a `TokenCredential` which I assume is only when it's a `ServiceClientCredential`.  But when it goes through this path, it never sets `credential` so it has no auth.  Also, options passed are ignored because it assumes the credential object are the options... so other parts break.  This part seems like a bug to me which I'll file.

File: package.nls.json
Comment (nturinski) [reply package.nls.json:13]: The command kind of changed.  It used to add a new image to the container app.  Each container app could have 4 images running simultaneously, so the verb deployed made more sense.

However, we decided that's a pretty edge use case, so we are now removing the old image and replacing it with the new image, so it functions more like an update.

**Comment by wwlorey** [comment]
> I don't think so. I can imagine the backlash of a bunch of users who got our token and were upset that the 0.10.0 update also broke all T1 SDK users which I imagine is practically everybody besides us. As far as I can tell, the access token is the exact same so Azure really shouldn't care as long as it's only using the default scope.

There are ways of avoiding fallout from removing `signRequest` especially now that we have a versioned API. But it's easier to leave it in and I think we should especially since it sounds like this is an SDK bug

File: src/tree/DaprTreeItem.ts
Comment (wwlorey) [inline src/tree/DaprTreeItem.ts:25]: Why is the description excluded here?

File: src/tree/ContainerAppTreeItem.ts
Comment (wwlorey) [inline src/tree/ContainerAppTreeItem.ts:152]: Has a bug been filed on the SDK for this?

**Comment by nturinski** [comment]
> > I don't think so. I can imagine the backlash of a bunch of users who got our token and were upset that the 0.10.0 update also broke all T1 SDK users which I imagine is practically everybody besides us. As far as I can tell, the access token is the exact same so Azure really shouldn't care as long as it's only using the default scope.
> 
> There are ways of avoiding fallout from removing `signRequest` especially now that we have a versioned API. But it's easier to leave it in and I think we should especially since it sounds like this is an SDK bug

Yeah, that's true.  We could have two separate token responses and users can pick which one that they need.  I'm not opposed to doing that either and it's not too late since users wouldn't know that we did that unless they had already adopted T2 packages.

We would just have to change all of our extensions to use this new credentials2 that we return.

File: src/tree/DaprTreeItem.ts
Comment (nturinski) [reply src/tree/DaprTreeItem.ts:25]: They changed where the dapr stuff lives now and I was too lazy to figure that out atm and fix it properly.

If they're going to change it on me, I'll just remove the tree entirely! (which I did.)

File: src/tree/ContainerAppTreeItem.ts
Comment (nturinski) [reply src/tree/ContainerAppTreeItem.ts:152]: It does now 😐

File: src/commands/chooseRevisionMode.ts
Comment (nturinski) [reply src/commands/chooseRevisionMode.ts:17]: sigh.... yeeeeeaah.  But I changed it because in this scope:

image

 Typescript forgets that I already determined that this is a `ContainerAppItem` and throws typing errors.

File: package.nls.json
Comment (nturinski) [reply package.nls.json:13]: Would you be okay with holding off a bit? @anthonychu and I are still kind of trying to figure out the best way to convey what this command does.  I was planning to rename everything (the file name, localize labels, etc.) but I didn't want to do that multiple times so was just waiting until we landed on something

Comment (wwlorey) [reply package.nls.json:13]: Sure thing. Definitely not a blocker for this 👍

File: src/tree/DaprTreeItem.ts
Comment (wwlorey) [reply src/tree/DaprTreeItem.ts:25]: Got it. I'm not sure how you're tracking issues like this one but it might be worth logging a GitHub issue on this repo  about dapr so comments like this don't get lost in the sauce

File: main.js
Comment (wwlorey) [inline main.js:22]: nit: revert this file

**Review by wwlorey — APPROVED** [review]

**Pull request merged**
